rend3-gltf/src/lib.rs

```diff
@@ -312,35 +337,60 @@ pub fn add_mesh_by_index<E: std::error::Error + 'static>(
     loaded: &LoadedGltfScene,
     mesh_index: usize,
     name: Option<&str>,
+    skin_index: Option<usize>,
     transform: Mat4,
 ) -> Result<Labeled<Object>, GltfLoadError<E>> {
     let mesh_handle = loaded
         .meshes
         .get(mesh_index)
         .ok_or(GltfLoadError::MissingMesh(mesh_index))?;
-    let primitives: Result<Vec<_>, GltfLoadError<_>> = mesh_handle
-        .inner
-        .primitives
-        .iter()
-        .map(|prim| {
-            let mat_idx = prim.material;
-            let mat = mat_idx
-                .map_or_else(
-                    || Some(&loaded.default_material),
-                    |mat_idx| loaded.materials.get(mat_idx).map(|m| &m.inner),
-                )
-                .ok_or_else(|| GltfLoadError::MissingMaterial(mat_idx.expect("Could not find default material")))?;
-            Ok(renderer.add_object(types::Object {
-                // TODO: Autodetect animation and spawn skeleton instead.
-                mesh_kind: ObjectMeshKind::Static(prim.handle.clone()),
-                material: mat.clone(),
-                transform,
-            }))
-        })
-        .collect();
+
+    let mut primitives: Vec<ObjectHandle> = vec![];
+    let mut skeletons: Vec<SkeletonHandle> = vec![];
+
+    let skin = if let Some(skin_index) = skin_index {
+        let skin = loaded
+            .skins
+            .get(skin_index)
+            .ok_or(GltfLoadError::MissingSkin(skin_index))?;
+        Some(skin)
+    } else {
+        None
+    };
+
+    for prim in mesh_handle.inner.primitives.iter() {
```

Comment:
The code here became a bit too involved for a nice iterator chain, so I refactored it into a good ol' imperative loop that pushes to vectors instead.

Reply:
nit: 
```suggestion
    for prim in &mesh_handle.inner.primitives {
```